sys/net/gnrc/routing/rpl/p2p/gnrc_rpl_p2p_dodag.c

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (C) 2015 Cenk Gündoğan <cnkgndgn@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author      Cenk Gündoğan <cnkgndgn@gmail.com>
+ */
+
+#include <string.h>
+#include "net/gnrc/rpl/structs.h"
+#include "net/gnrc/rpl/p2p.h"
+#include "net/gnrc/rpl/p2p_dodag.h"
+#include "net/gnrc/rpl/p2p_structs.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+gnrc_rpl_p2p_ext_t *gnrc_rpl_p2p_ext_new(gnrc_rpl_dodag_t *dodag)
+{
+    for (uint8_t i = 0; i < GNRC_RPL_P2P_EXTS_NUMOF; ++i) {
+        if (gnrc_rpl_p2p_exts[i].state == 0) {
+            gnrc_rpl_p2p_exts[i].state = 1;
+            gnrc_rpl_p2p_exts[i].dodag = dodag;
+            gnrc_rpl_p2p_exts[i].dro_delay = -1;
+            gnrc_rpl_p2p_exts[i].lifetime_sec = INT8_MIN;
```

Comment:
`GNRC_RPL_P2P_DEFAULT_LIFETIME` is used by the root node. All normal nodes will read the proper value (which is in our case also `GNRC_RPL_P2P_DEFAULT_LIFETIME`) from the RDO.